### ISSUE TITLE
Problems with connections closing and not re-establishing

### ISSUE THREAD

**Issue by JetForMe** [davidstump/SwiftPhoenixClient#214]
What is the default expected behavior in the presence of severe network errors? We're seeing a general behavior of SwiftPhoenixClient disconnecting and not re-connecting.

Interestingly, we see two separate behaviors. Our app connects to the same `ws` URL with two different `Socket`s, and then a single channel is created for each Socket, each with a different topic (one for "auction", one for "chat").

To reproduce the behavior, I ran the code on my iPhone and set the Network Link Conditioner to "Very Bad Network."

I see a lot of heartbeat timeouts on the `auction` channel, but never on the `chat` channel. Eventually I’ll see timeouts on the `transport,` and then things just close:

```
transport, heartbeat timeout. Attempting to re-establish connection
transport, close
```

What can we do here to try to keep re-establishing the connection? Does adding an error or close callback give me a good place to re-connect?

**Comment by dsrees** [comment]
The reconnection logic Should Just Work. When setting Network Link Conditioner to "Very Bad Network", are you able to reproduce the issue? 

Also, I would recommend only using a single `Socket` connection and then opening multiple channels through that socket.

**Comment by JetForMe** [comment]
Yes, it sometimes reproduces with Very Bad Network. I'm using `URLSessionWebSocketTask` (rather, I’m NOT using Starscream, so I assume it's using `URLSessionWebSocketTask`).

> Also, I would recommend only using a single Socket connection and then opening multiple channels through that socket.

Interesting, our back-end guys recommended separate sockets.

**Comment by JetForMe** [comment]
A little more investigation: It seems that generally, channels do reconnect. Socket disconnects are a little less reliable, although not 100% so. It would be nice if there was more logging of re-connection-in-the-face-of-errors attempts.
